docs/cow-protocol/reference/contracts/core/settlement.md

```diff
@@ -0,0 +1,198 @@
+---
+sidebar_position: 1
+---
+
+# GPv2Settlement
+
+## Architecture
+
+A settlement comprises of: 
+
+- A list of traded tokens with their corresponding price in the batch
+- A list of trades to execute
+- A list of interactions
+
+A solver monitors on-chain liquidity and on receiving [batch-auction instances](../../core/auctions/schema#instances-input) from the Protocol, it determines which orders can be matched, the clearing prices of the settlement, and what extra liquidity is necessary from the blockchain.
+
+Normally, all orders in a settlement are settled with _uniform clearing prices_, which means that every user receives the same price for the same token.
+
+If the transaction does not revert, each order will be executed during a settlement and the user will receive the desired token after the settlement.
+
+### Guarantees and Invariants
+
+The parameters of the order are verified for validity:
+
+- The signature for the order matches the user's address
+- The order is not expired
+- The order was not previously filled
+- The current prices are equal to or better than what is specified in the order
+
+### Trades
+
+Trades contain a description of the users' orders, a signature for verifying its validity, and the executed amount (for partially fillable orders).
+
+The contract decodes the order parameters from the trade.
+
+### Interactions
+
+Interactions allow solvers to execute arbitrary calls to any on-chain contract. Normally, they are used to interact with other on-chain liquidity providers, for example, to make a swap call to Uniswap.
+
+<details open>
+  <summary>Protocol fee collection</summary>
+  
+  Interactions are also used for accounting / bookkeeping purposes as well. As trades are executed, the Protocol collects a fee from each trade and stores this in the settlement contract (known as <em>internal buffers</em>). At regular intervals, the Protocol withdraws the fees from the settlement contract to the Protocol's treasury Safe.
+
+</details>
+
+:::tip
+
+Solvers are **NOT** able to directly access _user_ funds through interactions. There are strict guarantees enforced by the settlement contract over movement of user funds.
+
+:::
+
+:::caution
+
+Allowing interactions by solvers, creates the possibility of malicious solvers that can steal funds from the settlement contract. This is why the Protocol only allows interactions to be executed by whitelisted solvers, and requires that the solver post a bond to the Protocol before it can be whitelisted.
```

Comment:
Do we have a section explaining the process, reasoning and details for solver bonds in detail that we could link here?